soh/src/code/z_en_item00.c

```diff
@@ -1405,6 +1429,38 @@ void EnItem00_DrawHeartPiece(EnItem00* this, GlobalContext* globalCtx) {
     }
 }
 
+/**
+ * If the given drop type ID is a bomb, and bombchu drops are enabled, sometimes convert it into a bombchu.
+ * Returns the new drop type ID.
+ */
+s16 EnItem00_ConvertBombDropToBombchu(s16 dropId) {
+    if (!CVar_GetS32("gBombchuDrops", 0) ||
+        INV_CONTENT(ITEM_BOMBCHU) == ITEM_NONE ||
+        (dropId != ITEM00_BOMBS_A && dropId != ITEM00_BOMBS_B && dropId != ITEM00_BOMBS_SPECIAL)) {
+        return dropId;
+    }
```

Comment:
since this is only being called in one place, it might make sense to move some of this part out of this function

i think just removing `(dropId != ITEM00_BOMBS_A && dropId != ITEM00_BOMBS_B && dropId != ITEM00_BOMBS_SPECIAL)` from here and making the call be

```c
if (dropId == ITEM00_BOMBS_A || dropId == ITEM00_BOMBS_B || dropId == ITEM00_BOMBS_SPECIAL) {
  dropId = EnItem00_ConvertBombDropToBombchu(dropId);
}
```
might make it clearer